openvpn/middlewares/server/auth/middleware.go

```diff
@@ -23,21 +23,27 @@ import (
 	"strings"
 )
 
-// CredentialsChecker callback checks given auth primitives (i.e. customer identity signature / node's sessionId)
-type CredentialsChecker func(username, password string) (bool, error)
-
 type middleware struct {
-	checkCredentials CredentialsChecker
-	commandWriter    management.Connection
-	currentEvent     clientEvent
+	// TODO: consider implementing event channel to communicate required callbacks
+	credentialsValidator CredentialsValidator
+	sessionCleaner       SessionCleaner
+	commandWriter        management.Connection
```

Comment:
Not related to this PR, but this variable and class naming difference bugs me - why not call interface `management.CommandWriter` instead of `management.Connection`? Isn't `Connection` just a detail of implementation?